python/packages/autogen-ext/src/autogen_ext/agents/openai/_openai_agent.py

```diff
@@ -0,0 +1,724 @@
+import asyncio
+import json
+import logging
+import warnings
+from typing import Any, AsyncGenerator, Dict, List, Mapping, Optional, Sequence, Type, TypedDict, Union, cast
+
+from autogen_agentchat import EVENT_LOGGER_NAME
+from autogen_agentchat.agents import BaseChatAgent
+from autogen_agentchat.base import Response
+from autogen_agentchat.messages import (
+    AgentEvent,
+    BaseChatMessage,
+    ChatMessage,
+    HandoffMessage,
+    MultiModalMessage,
+    StopMessage,
+    TextMessage,
+    ToolCallSummaryMessage,
+)
+from autogen_core import CancellationToken, Component, ComponentModel, FunctionCall
+from autogen_core.models import UserMessage
+from autogen_core.tools import Tool
+from pydantic import BaseModel, Field
+
+from openai import AsyncAzureOpenAI, AsyncOpenAI
+
+# NOTE: We use the new Responses API, so ChatCompletion imports are not needed.
+
+event_logger = logging.getLogger(EVENT_LOGGER_NAME)
+
+
+def _convert_tool_to_function_schema(tool: Tool) -> Dict[str, Any]:
+    schema = tool.schema
+    parameters: Dict[str, object] = {}
+    if "parameters" in schema:
+        parameters = {
+            "type": schema["parameters"]["type"],
+            "properties": schema["parameters"]["properties"],
+        }
+        if "required" in schema["parameters"]:
+            parameters["required"] = schema["parameters"]["required"]
+
+    return {
+        "name": schema["name"],
+        "description": schema.get("description", ""),
+        "parameters": parameters,
+    }
+
+
+class OpenAIMessageContent(TypedDict):
+    type: str
+    text: str
+
+
+class OpenAIImageUrlContent(TypedDict):
+    url: str
+
+
+class OpenAIImageContent(TypedDict):
+    type: str
+    image_url: OpenAIImageUrlContent
+
+
+class OpenAIMessage(TypedDict):
+    role: str
+    content: Union[str, List[Union[OpenAIMessageContent, OpenAIImageContent]]]
+
+
+def _convert_message_to_openai_message(
+    message: Union[TextMessage, MultiModalMessage, StopMessage, ToolCallSummaryMessage, HandoffMessage],
+) -> OpenAIMessage:
+    """Convert an AutoGen message to an OpenAI message format."""
+    if isinstance(message, TextMessage):
+        if message.source == "user":
+            return {"role": "user", "content": str(message.content)}
+        elif message.source == "system":
+            return {"role": "system", "content": str(message.content)}
+        elif message.source == "assistant":
+            return {"role": "assistant", "content": str(message.content)}
+        else:
+            return {"role": "user", "content": str(message.content)}
+    elif isinstance(message, MultiModalMessage):
+        content_parts: List[Union[OpenAIMessageContent, OpenAIImageContent]] = []
+        for part in message.content:
+            if isinstance(part, TextMessage):
+                content_parts.append({"type": "text", "text": str(part.content)})
+            elif isinstance(part, ImageMessage):
+                image_url = str(part.content)
```

Comment:
According to https://github.com/microsoft/autogen/pull/6418/files#diff-81b975348a92efc2e2df60eade414fd6cddee0ff1e0bd8acd5481eeed70da0faR712-R715, `part.content` can be a base64 string.